backend/main.py

```
@@ -61,14 +61,29 @@ async def start_background_tasks(self):
+        retry_delay = 5
+
+        for attempt in range(max_retries):
+            try:
+                async with get_weaviate_client() as client:
+                    if await client.is_ready():
+                        logger.info("Successfully connected to Weaviate.")
+                        return
+
+                logger.warning("Weaviate client connected, but the service is not ready yet.")
+
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)